### PR TITLE
add link_test wrt new workflow

### PR DESCRIPTION
**Pull request by syamgk**
## What is the purpose of this change? What does it change?
I validated the link command with the latest master, and it was working for me while running manually on my Linux machine across dev cluster, raise this PR to validate it on CI

## Was the change discussed in an issue?
yes, #1492

fixes https://github.com/openshift/odo/issues/1616

## How to test changes?

running the end2end test locally or on a CI and validate the tests are passing

### CODE REVIEW

**Comment by mohammedzee1000** [comment]
/assign @amitkrout 
Can you take a look

File: tests/e2e/new_link_test.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		})
+
+		var _ = Context("when --app flag is used", func() {
+			It("create local openJDK component and push code and link to postgresql", func() {
+				helper.CopyExample(filepath.Join("source", "openjdk-sb-postgresql"), context)
+				importOpenJDKImage()
+
+				helper.CmdShouldPass("odo create java sb-app --app jdklinktest")
+				//TODO: verify that config was properly created
+				helper.CmdShouldPass("cat .odo/config.yaml")
```
Comment (amitkrout) [inline tests/e2e/new_link_test.go]: Unnecessary IMO. push will fail if config is not created properly anyway. BTW you are only checking the file, not verifying it.

File: tests/e2e/new_link_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+				helper.CmdShouldPass("odo push")
+
+				helper.CmdShouldPass("odo service create dh-postgresql-apb --app jdklinktest --plan dev -p postgresql_user=luke -p postgresql_password=secret -p postgresql_database=my_data -p postgresql_version=9.6")
```
Comment (amitkrout) [inline tests/e2e/new_link_test.go]: +1, Cool!!!

File: tests/e2e/new_link_test.go (hunk 1 above)
Comment (syamgk) [reply tests/e2e/new_link_test.go]: okey will remove it

File: tests/e2e/new_link_test.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+				helper.CmdShouldPass("odo url create --port 8080")
+				helper.CmdShouldPass("odo link dh-postgresql-apb -w --wait-for-target")
+				waitForCmdOut("odo service list | sed 1d", 1, true, func(output string) bool {
+					return strings.Contains(output, "dh-postgresql-apb") &&
+						strings.Contains(output, "ProvisionedAndLinked")
+				})
+
+				routeURL := determineRouteURL()
+				// Ping said URL
+				responseStringMatchStatus := matchResponseSubString(routeURL, "Spring Boot", 30, 1)
```
Comment (amitkrout) [inline tests/e2e/new_link_test.go]: +1

**Comment by girishramnani** [comment]
@amitkrout removed the file as requested

**Comment by amitkrout** [comment]
Will be tracked in #1700

**Comment by girishramnani** [comment]
why was this closed @amitkrout ?

**Comment by amitkrout** [comment]
Will be tracked in #1700 as per the new test workflow

**Pull request closed without merging**